cnn_framework/data.py

```diff
@@ -382,7 +382,7 @@ def prepare_full_train_data_from_file(datafile,
 
 
 @attr('helper')
-def split_test_from_train_and_val(X, y, extra_data=None, shuffle_seed=None, testing_ratio=0.1):
+def split_test_from_train_and_val(X, y, extra_data=None, shuffle_seed=0, testing_ratio=0.1):
```

Comment:
If this commit is necessary, can it be squashed with the previous one?

Reply:
It's not necessary.  But I feel it's better to make the code deterministic by default.  What do you think?

Reply:
Yeah, perhaps deterministic is better. Some of the methods/functions in `predictor.py` and `data.py` already set the seed explicitly when calling `split_test_from_train_and_val`, etc. If we want it to be fully deterministic, we would also have to set the seed for `train_model` in `cnn_model.py` and the Keras seed to that the weights are always initialized in the same way.

Reply:
I've added an option for users to set random seed as we discussed.